pkg/arvo/sys/arvo.hoon

```diff
@@ -156,6 +245,549 @@
+      |=  vax=vase
+      ^-  (pair move worm)
+      ~>  %mean.'bad-move'
+      =^  hip  sac  (~(nell wa sac) p.vax)
+      ?.  hip
+        ~>(%mean.'not a cell' !!)
+      =/  duc
+        ~>  %mean.'bad-duct'
+        ;;(duct -.q.vax)
+      ::
+      ::  yat: specialized ball vase
+      ::
+      =^  yat  sac  (~(spot wa sac) 3 vax)
+      =^  del  sac  (refine-ball yat)
+      [[duc del] sac]
+    ::  +refine-ball: ball from vase
+    ::
+    ++  refine-ball
+      |=  vax=vase
+      ^-  (pair ball worm)
+      ?+    q.vax
+          ~>  %mean.'bad-ball'
+          ~_  (sell vax)
+          !!
+      ::
+          [%give card]
+        ::  yed: vase containing card
+        ::  hil: card as maze
+        ::
+        =^  yed  sac  (~(spot wa sac) 3 vax)
+        =^  hil  sac  (refine-card yed)
+        [[%give hil] sac]
+      ::
+          [%pass wire=* vane=term card]
+        =/  =wire
+          ~>  %mean.'bad-wire'
+          ;;(wire wire.q.vax)
+        =/  vane
+          ~>  %mean.'bad-vane-label'
+          ?>  ((sane %tas) vane.q.vax)
+          vane.q.vax
+        ::
+        ::  yed: vase containing card
+        ::  hil: card as maze
+        ::
+        =^  xav  sac  (~(spot wa sac) 7 vax)
+        =^  yed  sac  (~(spot wa sac) 3 xav)
+        =^  hil  sac  (refine-card yed)
+        [[%pass wire vane hil] sac]
+      ::
+          [%slip vane=term card]
+        =/  vane
+          ~>  %mean.'bad-vane-label'
+          ?>  ((sane %tas) vane.q.vax)
+          vane.q.vax
+        ::
+        ::  yed: vase containing card
+        ::  hil: card as maze
+        ::
+        =^  xav  sac  (~(spot wa sac) 3 vax)
+        =^  yed  sac  (~(spot wa sac) 3 xav)
+        =^  hil  sac  (refine-card yed)
+        [[%slip vane hil] sac]
+      ::
+          [%hurl goof=^ ball=*]
+        =/  =goof
+          =/  mote  -.goof.q.vax
+          ?>  ?&  ?=(@ mote)
+                  ((sane %tas) mote)
+              ==
+          [mote ;;(tang +.goof.q.vax)]
+        ::
+        =^  bal  sac
+          =^  lab  sac  (~(spot wa sac) 7 vax)
+          $(vax lab)
+        ::
+        ?>  ?=(?(%pass %give) -.p.bal)
+        [[%hurl goof p.bal] sac]
+      ==
+    ::  +refine-card: card from vase
+    ::
+    ++  refine-card
+      |=  vax=vase
+      ^-  (pair maze worm)
+      ~>  %mean.'bad-card'
+      =^  hip  sac  (~(nell wa sac) p.vax)
+      ?>  hip
+      ?.  ?=(%meta -.q.vax)
+        ::
+        ::  for a non-meta card, the maze is the vase
+        ::
+        [[%& vax] sac]
+      ~>  %mean.'bad-meta'
+      ::
+      ::  tiv: vase of vase of card
+      ::  typ: vase of span
+      ::
+      =^  tiv  sac  (~(slot wa sac) 3 vax)
+      =^  hip  sac  (~(nell wa sac) p.tiv)
+      ?>  hip
+      =^  typ  sac  (~(slot wa sac) 2 tiv)
+      =.  sac  (~(neat wa sac) typ.vil [%& typ])
+      ::
+      ::  support for meta-meta-cards has been removed
+      ::
+      ?>  ?=(meta q.tiv)
+      [[%| q.tiv] sac]
+      ::
+      :: =/  mut
```

Comment:
I don't like this commented-out code.  We can always dig this out of git if we need to include it.

Reply:
I minor hedge on removing meta-meta cards; I wanted to show what it would be post-refactoring. Of course, meta-meta card reduction has been broken since a335974008d795ee5b9cb941b7f2404fe1069aeb, so it's not very risky to remove it. I'll delete the comment.